src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+    const lastInArray = this[this.length - 1];
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        resultString += `${separator}`;
+      } else {
+        resultString += this[i] + `${separator}`;
+      }
```

Comment:
```suggestion
      }
      
      resultString += this[i] + `${separator}`;
```

Reply:
It better works with 'else', so I will leave it

Reply:
> It better works with 'else', so I will leave it

Yeah, my bad, because in this case you need to use break in 'if' condition. You can leave it.